<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitReportReporter -->
<testsuite hostname="GSHYD-Q4DH444749" failures="2" tests="6" name="tests.SearchAndFilter" time="607.707" errors="1" timestamp="2025-04-10T17:39:32 IST" skipped="0">
  <testcase classname="tests.SearchAndFilter" name="searchDropDownAlphabeticalOrder" time="45.153"/>
  <system-out/>
  <testcase classname="tests.SearchAndFilter" name="validatePriceRangeFilter" time="256.188">
    <failure message="The following asserts failed:
	The Price is not in the correct range[3600-23700:40849 expected [true] but found [false],
	The Price is not in the correct range[3600-23700:40300 expected [true] but found [false],
	The Price is not in the correct range[3600-23700:26800 expected [true] but found [false],
	The Price is not in the correct range[3600-23700:36998 expected [true] but found [false],
	The Price is not in the correct range[3600-23700:91800 expected [true] but found [false],
	The Price is not in the correct range[3600-23700:39999 expected [true] but found [false],
	The Price is not in the correct range[3600-23700:28380 expected [true] but found [false],
	The Price is not in the correct range[3600-23700:55800 expected [true] but found [false],
	The Price is not in the correct range[3600-23700:28999 expected [true] but found [false],
	The Price is not in the correct range[3600-23700:27480 expected [true] but found [false],
	The Price is not in the correct range[3600-23700:91342 expected [true] but found [false]" type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
The Price is not in the correct range[3600-23700:40849 expected [true] but found [false],
The Price is not in the correct range[3600-23700:40300 expected [true] but found [false],
The Price is not in the correct range[3600-23700:26800 expected [true] but found [false],
The Price is not in the correct range[3600-23700:36998 expected [true] but found [false],
The Price is not in the correct range[3600-23700:91800 expected [true] but found [false],
The Price is not in the correct range[3600-23700:39999 expected [true] but found [false],
The Price is not in the correct range[3600-23700:28380 expected [true] but found [false],
The Price is not in the correct range[3600-23700:55800 expected [true] but found [false],
The Price is not in the correct range[3600-23700:28999 expected [true] but found [false],
The Price is not in the correct range[3600-23700:27480 expected [true] but found [false],
The Price is not in the correct range[3600-23700:91342 expected [true] but found [false]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:47)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:31)
at pages.HomePage.validatePriceRangeFilter(HomePage.java:233)
at tests.SearchAndFilter.validatePriceRangeFilter(SearchAndFilter.java:42)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1570)
]]>
    </failure>
  </testcase> <!-- validatePriceRangeFilter -->
  <system-out/>
  <testcase classname="tests.SearchAndFilter" name="searchProductAndValidate" time="57.146"/>
  <system-out/>
  <testcase classname="tests.SearchAndFilter" name="validateSearchSuggestions" time="32.458">
    <error message="stale element reference: stale element not found in the current frame
  (Session info: chrome=135.0.7049.42)
For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#stale-element-reference-exception
Build info: version: &#039;4.10.0&#039;, revision: &#039;c14d967899&#039;
System info: os.name: &#039;Mac OS X&#039;, os.arch: &#039;aarch64&#039;, os.version: &#039;15.4&#039;, java.version: &#039;22.0.2&#039;
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [f6388c029b7b32b27370977c1ce5d4df, getElementText {id=f.8B634F1D4CAA57B55AF3861AB8A7A141.d.CA8676DB00C6491106AC32EC98776932.e.78}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 135.0.7049.42, chrome: {chromedriverVersion: 135.0.7049.84 (6c019e560019..., userDataDir: /var/folders/bn/_pv1z4w536q...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:60679}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:60679/devtoo..., se:cdpVersion: 135.0.7049.42, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Element: [[ChromeDriver: chrome on mac (f6388c029b7b32b27370977c1ce5d4df)] -&gt; xpath: //div[contains(@id,&#039;sac-suggestion&#039;)]//div[contains(@class,&#039;suggestion-ellipsis-direction&#039;)]]
Session ID: f6388c029b7b32b27370977c1ce5d4df" type="org.openqa.selenium.StaleElementReferenceException">
      <![CDATA[org.openqa.selenium.StaleElementReferenceException: stale element reference: stale element not found in the current frame
  (Session info: chrome=135.0.7049.42)
For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#stale-element-reference-exception
Build info: version: '4.10.0', revision: 'c14d967899'
System info: os.name: 'Mac OS X', os.arch: 'aarch64', os.version: '15.4', java.version: '22.0.2'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [f6388c029b7b32b27370977c1ce5d4df, getElementText {id=f.8B634F1D4CAA57B55AF3861AB8A7A141.d.CA8676DB00C6491106AC32EC98776932.e.78}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 135.0.7049.42, chrome: {chromedriverVersion: 135.0.7049.84 (6c019e560019..., userDataDir: /var/folders/bn/_pv1z4w536q...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:60679}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:60679/devtoo..., se:cdpVersion: 135.0.7049.42, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Element: [[ChromeDriver: chrome on mac (f6388c029b7b32b27370977c1ce5d4df)] -> xpath: //div[contains(@id,'sac-suggestion')]//div[contains(@class,'suggestion-ellipsis-direction')]]
Session ID: f6388c029b7b32b27370977c1ce5d4df
at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.createException(W3CHttpResponseCodec.java:199)
at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:132)
at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:51)
at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:191)
at org.openqa.selenium.remote.service.DriverCommandExecutor.invokeExecute(DriverCommandExecutor.java:196)
at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:171)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:531)
at org.openqa.selenium.remote.RemoteWebElement.execute(RemoteWebElement.java:242)
at org.openqa.selenium.remote.RemoteWebElement.getText(RemoteWebElement.java:192)
at pages.HomePage.validateSearchSuggestions(HomePage.java:295)
at tests.SearchAndFilter.validateSearchSuggestions(SearchAndFilter.java:58)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1570)
]]>
    </error>
  </testcase> <!-- validateSearchSuggestions -->
  <system-out/>
  <testcase classname="tests.SearchAndFilter" name="validatePriceSorting" time="17.940"/>
  <system-out/>
  <testcase classname="tests.SearchAndFilter" name="validateFilters" time="198.822">
    <failure message="The following asserts failed:
	Storage Filter is showing incorrect data expected [true] but found [false],
	Found a discount below 25%: 21% expected [true] but found [false],
	Found a discount below 25%: 21% expected [true] but found [false],
	Some delivery-recipe divs do not contain the Prime icon!,
	Some delivery-recipe divs do not contain the Prime icon!,
	Some delivery-recipe divs do not contain the Prime icon!,
	Some delivery-recipe divs do not contain the Prime icon!,
	Some delivery-recipe divs do not contain the Prime icon!,
	Some delivery-recipe divs do not contain the Prime icon!,
	Some delivery-recipe divs do not contain the Prime icon!,
	Some delivery-recipe divs do not contain the Prime icon!,
	Some delivery-recipe divs do not contain the Prime icon!,
	Some delivery-recipe divs do not contain the Prime icon!,
	Some delivery-recipe divs do not contain the Prime icon!,
	Some delivery-recipe divs do not contain the Prime icon!,
	Some delivery-recipe divs do not contain the Prime icon!" type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: The following asserts failed:
Storage Filter is showing incorrect data expected [true] but found [false],
Found a discount below 25%: 21% expected [true] but found [false],
Found a discount below 25%: 21% expected [true] but found [false],
Some delivery-recipe divs do not contain the Prime icon!,
Some delivery-recipe divs do not contain the Prime icon!,
Some delivery-recipe divs do not contain the Prime icon!,
Some delivery-recipe divs do not contain the Prime icon!,
Some delivery-recipe divs do not contain the Prime icon!,
Some delivery-recipe divs do not contain the Prime icon!,
Some delivery-recipe divs do not contain the Prime icon!,
Some delivery-recipe divs do not contain the Prime icon!,
Some delivery-recipe divs do not contain the Prime icon!,
Some delivery-recipe divs do not contain the Prime icon!,
Some delivery-recipe divs do not contain the Prime icon!,
Some delivery-recipe divs do not contain the Prime icon!,
Some delivery-recipe divs do not contain the Prime icon!
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:47)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:31)
at pages.HomePage.addFiltersAndValidateResults(HomePage.java:359)
at tests.SearchAndFilter.validateFilters(SearchAndFilter.java:64)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1570)
]]>
    </failure>
  </testcase> <!-- validateFilters -->
  <system-out/>
</testsuite> <!-- tests.SearchAndFilter -->
